src/base/Cellar.sol

```diff
@@ -1069,6 +1116,58 @@ contract Cellar is ERC4626, Ownable, Multicall {
 
     // ========================================= FEES LOGIC =========================================
 
+    /**
+     * @notice Calculates how many assets Strategist would earn performance fees
+     * @param _totalAssets uint256 value of the total assets in the cellar
+     * @return feeInAssets amount of assets
+     * @return currentSharePrice current price of shares, used by _takePerformanceFee
+     */
+    function _previewPerformanceFees(uint256 _totalAssets)
+        internal
+        view
+        returns (uint256 feeInAssets, uint256 currentSharePrice)
+    {
+        if (feeData.performanceFee == 0 || _totalAssets == 0) return (0, 0);
+        currentSharePrice = _convertToAssets(10**decimals, _totalAssets);
+        if (feeData.sharePriceHighWatermark < currentSharePrice) {
+            //find how many assets make up the fee
+            uint256 yield = ((currentSharePrice - feeData.sharePriceHighWatermark) * totalSupply) / 10**decimals;
+            feeInAssets = yield.mulDivDown(feeData.performanceFee, 1e18);
```

Comment:
Optimize this by storing `feeData.performanceFee` and `feeData.sharePriceHighWatermark` and reading from that instead of reading from state redundantly.